source/develop/release-management/features/infra/link-following.html.md

```diff
@@ -0,0 +1,114 @@
+
+The href attribute of the returned entity/entities is used to fetch it/them. This is done by crawling along the API tree to the context where the information can be retrieved. 
+
+Crawling along the API tree is done using <code>ServiceTree</code> class, which was enhanced to provide this kind of service (https://gerrit.ovirt.org/#/c/76077/).
+
+After the information has been retrieved it needs to be set in the original entity (for example, <code>vm.setCluster(cluster)</code>. Getting from follow=cluster,... to vm.setCluster() is done by string manipulation, but would ideally be done using an annotation (similarly to get)
+
+All business-logic described above is placed in: <code>follow(ActionableResource entity)</code> in <code>BaseBackendResource</code> - the parent class for all API resources.
+
+This method is transparently invoked. To achieve transparent invocation, generated Jax-rs interfaces were made to include <code>doGet(), doList()</code> methods, and jax-rs <code>@GET</code> annotations was moved to them, making GET requests land there. These methods have a default implementation which invokes <code>follow()</code>:
+
+<code>@GET</code>
+<code>default public Host doGet(String id) {</code>
+<code>   Host host = get();</code>
+<code>   follow(host);</code>
+<code>   return host;</code>
+<code>}</code>
+
+This <code>follow()</code> method is final, purposely forbidding developers from overriding it. But as part of its flow it invokes another signature of <code>follow()</code>. This method is the entry point for developers who need to add specific following logic.
+
+For cases like <code>GET ...api/vms?follow="a,b,c"</code>, when developer adds specific logic for following a, his code should include:
+
+<ol>
+<li>Check if a exists in follow url parameter.</li>
+<li>Follow a and insert results into the vm</li>
+<li>Mark a as followed.</li>
+</ol>
+
+Example: get Vms with Nics.
+• Class: <code>BackendVmsResource</code>
+• Override <code>follow( ActionableResource entity, LinksTreenode linksTree )</code>
+Decide whether to intervene: <coce>Boolean exists = linksTree.pathExists(“nics”)</code>
+Run engine query: <code>getBackendCollection(VdcQueryType.GetVmsWithNics). </code>
+Map and set data
+Mark followed=true: <code>linksTree.markAsFollowed(“nics”)</code>
+
+## Prerequisites
+
+None
+
+## Limitations
+
+The generic solution is naive and costly in terms of performance. Frequent cases requiring link-following should be handled manually (the infrastructure provides an intervention point)
+
+## Benefit to oVirt
+
+This features saves the need from client-side scripts for following links and enables running designated queries for common scenarios (e.g: get vms+nics+disks)
```

Comment:
Suggest:
"this feature" or "these features"

"links, and enables

Add full stop at end of sentence.